File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.py
# Last updated: 8/1/2025, 10:42:16 PM
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle:
            return 0  
        for i in range(len(haystack) - len(needle) + 1):
            
            if haystack[i:i+len(needle)] == needle:
                return i
        
        return -1  

File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.java
// Last updated: 8/1/2025, 10:42:10 PM
class Solution {
    public int strStr(String haystack, String needle) {
        if(haystack.length()<needle.length()) return -1;

        if(needle.length()==0) return 0;

        int p1=0;
        int p2=needle.length()-1;
        while(p2<haystack.length()){
            if(haystack.substring(p1,p2+1).equals(needle)){
                return p1;
            }
            p1++;
            p2++;
         }
         return -1;
    }
}